File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.java
import java.util.*;

public class Solution {
    public int waysToSplitArray(int[] nums) {
        int n = nums.length;
        long sum = 0;
        for (int num : nums) sum += num;

        long lsum = 0, rsum = 0;
        int splitCount = 0;

        for (int i = 0; i < n - 1; i++) {
            lsum += nums[i];
            rsum = sum - lsum;
            if (lsum >= rsum) splitCount++;
        }
        return splitCount;
    }
}

File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int n = nums.size();
        long long sum = 0;
        for(auto &n:nums)sum+=n;
        long long lsum=0,rsum=0;
        int splitc=0;
        for(int i =0; i<n-1;i++){
            lsum +=nums[i],rsum = sum-lsum;
            if(lsum>=rsum)splitc++;
        }
        return splitc;
    }
};